### PR TITLE
Added Filter Parameter

### LINKED ISSUE #279: Nested imports have wrong path
```
/component1/index.sss
/component2/index.sss
```

```
// component1
@import '@scope/module/file.sss'
```

```
// component2
@import '../component1/index.sss'
```

The `component2` will throw an error:

```
Module build failed: Error: ENOENT: no such file or directory, open '/Users/FezVrasta/app/component2/@scope/components1/index.sss'
```

### LINKED ISSUE #297: Filtering imports
I'd like to be able to filter (with whitelist/blacklist/custom-function) the files that are imported by postcss-import so that certain files can be resolved by the webpack css-loader instead (eg. for unique modules). I tried passing in a `load` function but I may be doing it incorrectly:
```
require('postcss-import')({
	load: (function () {

		const loadContent = require('postcss-import/lib/load-content');

		return function (filename, options) {
			if (filename.match(/module-for-css-loader/)) {
				return `@import "~module-for-css-loader";`;
			}

			return loadContent(filename, options);
		};
	})(),
}),
```

### PR DESCRIPTION
**Pull request by 0xcaff**
This allows for conditionally processing imports based on their URLs.

Closes #297, #254, #299, #259 

Docs and tests have been added.

This PR will help solve the following issues:

https://github.com/postcss/postcss-import/issues/179
https://github.com/css-modules/css-modules/pull/65

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,6 +110,14 @@ Checkout the [tests](test) for more examples.
 
 ### Options
 
+### `filter`
+Type: `string => boolean`  
+Default: `() => true`
+
+Only transforms imports for which the test function returns true. Imports for
+which the test function returns false will be left as is. The argument to the
+function is the url from the import (`@import (url)`).
```
Comment (RyanZim) [inline README.md]: Revise to:

> Type: `Function` 
> Default: `() => true`
> 
> Only transform imports for which the test function returns `true`. Imports for
which the test function returns `false` will be left as is. Function gets the path to import as an argument and should return a boolean.

**Comment by 0xcaff** [comment]
Why would this be a concern? Import statements must precede all other css statements except `@charset`? I don't see any other way this could work. I think this behavior is expected and should just be documented.

**Comment by RyanZim** [comment]
CSS is order-sensitive; but since imports must be at the top, we don't really have alternatives here.

> should just be documented

Probably right here.

**Comment by 0xcaff** [comment]
I've added the note to the readme about import order. Anything else?

**Comment by RyanZim** [comment]
Sorry about the wait here, not sure how this slipped through the cracks. I'll try to merge and release soon.

**Pull request merged**